src/routing_core.rs

```diff
@@ -73,6 +73,26 @@ pub enum State {
     Terminated,
 }
 
+/// ExpectedConnection.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug, Clone)]
+#[allow(unused)]
+pub enum ExpectedConnection {
+    /// ConnectRequest sent by peer.
+    Request(::messages::ConnectRequest),
+    /// ConnectResponse in response to a ConnectRequest sent by peer.
+    Response(::messages::ConnectResponse),
+}
+
+/// UnknownConnection.
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug, Clone)]
+#[allow(unused)]
+pub struct UnknownConnection {
```

Comment:
This is not the construction needed;

as in rfc-0011
RoutingCore {
  unknown_connections: ExpirationMap<::crust::Connection, Option<::direct_messages::Hello>>,